pilot/pkg/config/monitor/monitor.go

```diff
@@ -15,15 +15,13 @@
 package monitor
 
 import (
-	"os"
 	"reflect"
 	"strings"
-	"syscall"
+	"time"
 
+	"github.com/fsnotify/fsnotify"
```

Comment:
Were there issues with `istio.io/pkg/appsignals`? Should we replace our usage of this library everywhere?

Reply:
Yes. The package will drop events - if the reader is busy during an update it will just throw the event away. So if we apply many configs at once, we lose the last few usually. it also relies on some really roundabout notification method. With this implementation we go from file event -> channel. With appsignals, we go from file event -> channel -> os.SIGUSR1 -> watcher for os.SIGUSR1. appsignals is also as far as I can tell completely global so if we had multiple watchers any update seems to trigger all of them.

Reply:
@louiscryan IIRC you created `appsignals` because of issues with another library. Were you using `fsnotify` previously?

Reply:
Just circling back to this. Discussed offline and it sounds like the new version of fsnotify has replaced the old (buggy) version. I suspect it's fine to proceed.